src/gateways/googleAnalyticsClient.ts

```diff
@@ -106,7 +107,7 @@ class GoogleAnalyticsClient {
       return;
     }
 
-    if (this.isReadyForCommands()) {
+    if (this.isReadyForCommands() && !Cookies.get('ANALYTICS_OPT_OUT')) {
```

Comment:
I think this check should be moved directly into `isReadyForCommands`
and about the cookie name - we're used to use `constants.ts` file for such things

you should also add a unit test for this to check if `window.ga` is not called if cookie is set and that it is added to the `pendingCommands` instead

Reply:
I'm not sure where where `constants.ts` should be created. Like `src/constants.ts`, `src/gateways/constants.ts` or `src/app/constants.ts`? It doesn't seem right to me.

Reply:
We tend to put them in the folder where are functions that will use them so in this case I think `src/gateways/constants.ts ` will be okay